bundle/Form/DataMapper.php

```diff
@@ -0,0 +1,151 @@
+            } else {
+                $form->setData($form->getConfig()->getData());
+            }
+        }
+    }
+
+    public function mapFormsToData($forms, &$data): void
+    {
+        if (null === $data) {
+            return;
+        }
+
+        if (!is_array($data) && !is_object($data)) {
+            throw new UnexpectedTypeException($data, 'object, array or empty');
+        }
+
+        foreach ($forms as $form) {
+            $propertyPath = $form->getPropertyPath();
+            $config = $form->getConfig();
+
+            // Write-back is disabled if the form is not synchronized (transformation failed),
+            // if the form was not submitted and if the form is disabled (modification not allowed)
+            if (
+                null === $propertyPath
+                || !$config->getMapped()
+                || !$form->isSubmitted()
+                || !$form->isSynchronized()
+                || $form->isDisabled()
+            ) {
+                continue;
+            }
+
+            // If $data is out ContentCreateStruct, we need to map it to the corresponding field
+            // in the struct
+            if ($data instanceof DataWrapper) {
+                /* @var $data \Netgen\Bundle\InformationCollectionBundle\Form\DataWrapper */
+                $this->mapFromForm($form, $data, $propertyPath);
+
+                continue;
+            }
+
+            // If the field is of type DateTime and the data is the same skip the update to
+            // keep the original object hash
+            if (
+                $form->getData() instanceof DateTime
+                && $form->getData() === $this->propertyAccessor->getValue($data, $propertyPath)
+            ) {
+                continue;
+            }
+
+            // If the data is identical to the value in $data, we are
+            // dealing with a reference
+            if (
+                is_object($data)
+                && $config->getByReference()
+                && $form->getData() === $this->propertyAccessor->getValue($data, $propertyPath)
+            ) {
+                continue;
+            }
+
+            $this->propertyAccessor->setValue($data, $propertyPath, $form->getData());
+        }
+    }
+
+    /**
+     * Maps data from Ibexa Platform structure to the form.
+     */
+    abstract protected function mapToForm(
+        FormInterface $form,
+        DataWrapper $data,
+        PropertyPathInterface $propertyPath
+    ): void;
+
+    /**
+     * Maps data from form to the Ibexa Platform structure.
+     */
+    abstract protected function mapFromForm(
+        FormInterface $form,
+        DataWrapper $data,
+        PropertyPathInterface $propertyPath
+    ): void;
+
+    /**
+     * Returns if the update should be skipped for empty value.
+     *
+     * @param mixed $value
+     */
+    protected function shouldSkipForEmptyUpdate(FormInterface $form, $value, string $fieldDefinitionIdentifier): bool
```

Comment:
This is not needed, as it's used for updating Content and User in the original repo, which we don't need and won't port here.